src/api/app/controllers/search_controller.rb

```diff
@@ -275,4 +288,16 @@ def find_items(what, predicate)
     raise IllegalXpathError, "Error found searching elements '#{what}' with xpath predicate: '#{predicate}'.\n\n" \
                              "Detailed error message from parser: #{e.message}"
   end
+
+  def search_results_exceed_configured_limit?(matches)
+    config_limit = CONFIG['limit_for_search_results']
+    return false if config_limit.blank?
+
+    params_limit = params[:limit].present? && params[:limit] =~ /\A\d+\z/ ? params[:limit].to_i : nil
+
+    returned_results = params_limit.present? && params_limit < matches ? params_limit : matches
+    return false if returned_results <= config_limit
+
+    true
+  end
```

Comment:
I think you need to untangle `params_limit` from here and do this somewhere else. It has little to do with the configured limit IMHO.

Reply:
`params_limit` has to be taken in account with matches and the configured limit. I'll try to illustrate that with two examples. First example:
- matches: 100
- configured limit: 50
- `params_limit`: 10

In this case, the search results (not returned by matches, but by `params_limit`) don't exceed the configured limit.

In this other example:
- matches: 100
- configured limit: 50
- `params_limit`: 70

In this case, the search results (not returned by matches, but by `params_limit`) **do** exceed the configured limit. That's why `params_limit` is included here, and has to do with the configured limit.

Reply:
Sure in the case of `params_limit` existing, `params_limit` becomes `matches` right? That doesn't mean you have to do it in this method though....

Reply:
> Sure in the case of `params_limit` existing, `params_limit` becomes `matches` right? That doesn't mean you have to do it in this method though....

Not always. I still think we have to do it in this method. If the config limit is not defined, it is irrelevant if `params_limit` is defined or not. In other words: `params_limit` becomes `matches` only when config limit is defined. We should not make `matches` becoming `params_limit` if config limit is not defined.

We could extract some logic, from inside the method to some other method. But I don't think the result would be more clear. I would prefer to leave the method as it is.

Obviously, I could be missing something. Please provide an alternative if you have something better in mind.